techniques/applications/yumPackageManagerRepositories/1.0/yum-repositories-management.st

```diff
@@ -0,0 +1,81 @@
+      "filename[${index}]" string => "/etc/yum.repos.d/rudder-${repo_id[${index}]}.repo",
+                                if => "repo_entry_absent_${index}";
+      
+      "canonified_filename[${index}]" string => canonify("${filename[${index}]}");
+
+  classes:
+      "delete_repo_${index}" expression => strcmp("${repo_action$[${index}]}", "delete");
+
+    pass1::
+      "repo_entry_absent_${index}" expression => strcmp("${filename[${index}]}", "");
+
+
+    any::
+      "pass3" expression => "pass2";
+      "pass2" expression => "pass1";
+      "pass1" expression => "any";
+
+
+  methods:
+    pass3::
+      # create or enforce entries
+      # if file doesn't exist yet, we need to precreate it with an empty section, and a blank line
+      "create file ${index}"       usebundle => file_content("${filename[${index}]}", "[${repo_id[${index}]}]
+", "false"),
+                                          if => "!delete_repo_${index}.repo_entry_absent_${index}";
+      "set repo name ${index}"     usebundle => file_key_value_present_in_ini_section("${filename[${index}]}", "${repo_id[${index}]}", "name", "${repo_name[${index}]}"),
+                                          if => "!delete_repo_${index}";
+      "set repo url ${index}"      usebundle => file_key_value_present_in_ini_section("${filename[${index}]}", "${repo_id[${index}]}", "baseurl", "${repo_url[${index}]}"),
+                                          if => "!delete_repo_${index}";
+      "set repo enabled ${index}"  usebundle => file_key_value_present_in_ini_section("${filename[${index}]}", "${repo_id[${index}]}", "enabled", "${repo_enabled[${index}]}"),
+                                          if => "!delete_repo_${index}";
+      "set repo gpgcheck ${index}" usebundle => file_key_value_present_in_ini_section("${filename[${index}]}", "${repo_id[${index}]}", "gpgcheck", "${repo_gpg_check[${index}]}"),
+                                          if => "!delete_repo_${index}";
+      "set repo gpg uri ${index}"  usebundle => file_key_value_present_in_ini_section("${filename[${index}]}", "${repo_id[${index}]}", "gpgkey", "${repo_gpg_uri[${index}]}"),
+                                          if => "!delete_repo_${index}";
+
+
+      # report
+      # Using old_class_prefix to avoid having to copy all classes. Drawback is
+      # if there are several repos managed in the same file, there will be
+      # classes colision
+      "report for modification"    usebundle => rudder_common_reports_generic_index("Package sources (Yum)", "file_key_value_present_in_ini_section_${canonified_filename[${index}]}", "${trackingkey[${index}]}", "Repositories", "${repo_id[${index}]}", "Setting repository ${repo_name[${index}]} with id ${repo_id[${index}]} ", "${index}"),
+                                          if => "!delete_repo_${index}";
+                                          
+      # Deletion of repo
+      "delete repo ${index}"      usebundle => file_absent("${filename[${index}]}"),
```

Comment:
this is overzealous as it may delete several repos if they are in the same file

Reply:
i could make a file promise, that would select the region, and delete all lines matching *, plus deleting the region.
but it may leave nearly empty files

So another idea would be to count the section in a file, with a grep -rc '^\[.*\]$' /etc/yum.repos.d/
if count is 1, then delete the file, else clean it

this would probably be another bundle